ext/spl/spl_directory.c

```diff
@@ -1530,10 +1528,16 @@ PHP_METHOD(RecursiveDirectoryIterator, getChildren)
 		RETURN_THROWS();
 	}
 
-	ZVAL_LONG(&zflags, intern->flags);
-	ZVAL_STR_COPY(&zpath, intern->file_name);
-	spl_instantiate_arg_ex2(Z_OBJCE_P(ZEND_THIS), return_value, &zpath, &zflags);
-	zval_ptr_dtor(&zpath);
+	zval params[2];
+	ZVAL_STR_COPY(&params[0], intern->file_name);
+	ZVAL_LONG(&params[1], intern->flags);
+	/* Instantiate object and call constructor */
+	zend_result is_initialized = object_init_with_constructor(return_value, Z_OBJCE_P(ZEND_THIS), 2, params, NULL);
+	zval_ptr_dtor(&params[0]);
```

Comment:
You could use zval_ptr_dtor_str here. Not sure how hot the "getChildren" method is.

Reply:
I think this can still in theory return a non string variable because of the tentative return types.

Reply:
I'm dumb, I somehow thought we were freeing the retval of the method call...